command/agent/testagent.go

```diff
@@ -0,0 +1,278 @@
+package agent
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/consul/consul"
+	"github.com/hashicorp/consul/consul/structs"
+	"github.com/hashicorp/consul/logger"
+	"github.com/hashicorp/consul/testutil/retry"
+	"github.com/hashicorp/consul/types"
+	"github.com/hashicorp/consul/version"
+	uuid "github.com/hashicorp/go-uuid"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano()) // seed random number generator
```

Comment:
It's shared across projects so there's some justification, though it could be paste-able pretty easily. There's some nuance in doing it correctly (doesn't matter for unit tests), so that probably tips in favor vs. pasting.